luigi/parameter.py

```diff
@@ -219,6 +219,8 @@ def parse(self, x):
         :param str x: the value to parse.
         :return: the parsed value.
         """
+        if not isinstance(x, basestring) or not isinstance(x, str):
```

Comment:
I can only think of the following:

```python
def parse(self, x):
     try:  # try python2 syntax
          if not isinstance(x, basestring):
               warnings.warn(...)
      except:  # if py2 syntax fails, try py3
          if not isinstance(x, str):
               warnings.warn(...)
      return x
```

But this repeats `warnings.warn(...)` :disappointed: